platform/saivpp/vpplib/vppxlate/SaiVppXlate.c

```diff
@@ -1927,6 +1989,144 @@ int vpp_acl_del (uint32_t acl_index)
+        }
+    } else {
+        VPP_UNLOCK();
+        return -EINVAL;
+    }
+    mp->is_add = is_bind;
+    mp->tunterm_acl_index= htonl(tunterm_index);
+
+    S (mp);
+    W (ret);
+
+    VPP_UNLOCK();
+
+    return ret;
+}
+
+
+int vpp_tunterm_acl_del (uint32_t tunterm_index)
+{
+    vat_main_t *vam = &vat_main;
+    vl_api_tunterm_acl_del_t *mp;
+    int ret;
+
+    VPP_LOCK();
+
+    __plugin_msg_base = tunterm_msg_id_base;
+    M (TUNTERM_ACL_DEL, mp);
+
+    mp->tunterm_acl_index= htonl(tunterm_index);
+
+    S (mp);
+    W (ret);
+
+    VPP_UNLOCK();
+
+    return ret;
+}
+
+int vpp_tunterm_acl_add_replace (uint32_t *tunterm_index, uint32_t count, vpp_tunterm_acl_t *in_acl)
+{
+    u32 idx;
+    vat_main_t *vam = &vat_main;
+    vpp_ip_addr_t *addr;
+    tunterm_acl_rule_t *in_rule;
+    vl_api_address_t *api_addr;
+    vl_api_tunterm_acl_rule_t *vpp_rule;
+    vl_api_tunterm_acl_add_replace_t *mp;
+    int ret;
+
+    init_vpp_client();
+
+    VPP_LOCK();
+
+    __plugin_msg_base = tunterm_msg_id_base;
+
+    M2 (TUNTERM_ACL_ADD_REPLACE, mp, count*sizeof(vl_api_tunterm_acl_rule_t));
+
+    mp->count = htonl(count);
+    mp->tunterm_acl_index = htonl(*tunterm_index);
+
+    for (idx = 0; idx < count; idx++) {
+        in_rule = &in_acl->rules[idx];
+        vpp_rule = &mp->r[idx];
+
+        addr = &in_rule->dst_prefix;
+        api_addr = &vpp_rule->dst;
+
+        if (!vpp_to_vl_api_ip_addr(api_addr, addr)) {
+            SAIVPP_WARN("Unknown protocol in tunterm acl destination prefix");
+            VPP_UNLOCK();
+            return -EINVAL;
+        }
+
+        if (addr->sa_family == AF_INET6) {
+            mp->is_ipv6 = true;
+        } else {
+            mp->is_ipv6 = false;
+        }
+
+        if (in_rule->hwif_name) {
+            u32 idx;
+            idx = get_swif_idx(vam, in_rule->hwif_name);
+            if (idx != (u32) -1) {
+                vpp_rule->path.sw_if_index = htonl(idx);
+            } else {
+                SAIVPP_ERROR("Unable to get sw_index for %s\n", in_rule->hwif_name);
+                VPP_UNLOCK();
+                return -EINVAL;
+            }
+        } else {
+            SAIVPP_ERROR("No hwif_name provided.\n");
+            VPP_UNLOCK();
+            return -EINVAL;
+        }
+
+        if (in_rule->ip_protocol == 1) {
+            vpp_rule->path.proto = htonl(FIB_API_PATH_NH_PROTO_IP4);
+            memcpy(vpp_rule->path.nh.address.ip4, &in_rule->next_hop_ip.addr.ip4.sin_addr.s_addr, sizeof(vpp_rule->path.nh.address.ip4)); 
+        } else if (in_rule->ip_protocol == 2) {
+            vpp_rule->path.proto  = htonl(FIB_API_PATH_NH_PROTO_IP6);
+            memcpy(vpp_rule->path.nh.address.ip6, &in_rule->next_hop_ip.addr.ip6.sin6_addr.s6_addr, sizeof(vpp_rule->path.nh.address.ip6));
+        } else {
+            SAIVPP_WARN("Unknown protocol in next hop prefix");
+            VPP_UNLOCK();
+            return -EINVAL;
+        }
+    }
+    mp->context = store_ptr(tunterm_index);
```

Comment:
why do we need it in context? Is it already passed in with tunterm_acl_index?

Reply:
In create case the VPP tunterm sw_index will be returned here